core/src/main/java/org/apache/iceberg/io/FanoutPositionOnlyDeleteWriter.java

```diff
@@ -49,21 +51,32 @@ public FanoutPositionOnlyDeleteWriter(
       OutputFileFactory fileFactory,
       FileIO io,
       long targetFileSizeInBytes) {
+    this(writerFactory, fileFactory, io, targetFileSizeInBytes, DeleteGranularity.PARTITION);
+  }
+
+  public FanoutPositionOnlyDeleteWriter(
+      FileWriterFactory<T> writerFactory,
+      OutputFileFactory fileFactory,
+      FileIO io,
+      long targetFileSizeInBytes,
+      DeleteGranularity granularity) {
     this.writerFactory = writerFactory;
     this.fileFactory = fileFactory;
     this.io = io;
     this.targetFileSizeInBytes = targetFileSizeInBytes;
+    this.granularity = granularity;
     this.deleteFiles = Lists.newArrayList();
     this.referencedDataFiles = CharSequenceSet.empty();
   }
 
   @Override
   protected FileWriter<PositionDelete<T>, DeleteWriteResult> newWriter(
       PartitionSpec spec, StructLike partition) {
-    FileWriter<PositionDelete<T>, DeleteWriteResult> delegate =
-        new RollingPositionDeleteWriter<>(
-            writerFactory, fileFactory, io, targetFileSizeInBytes, spec, partition);
-    return new SortingPositionOnlyDeleteWriter<>(delegate);
+    return new SortingPositionOnlyDeleteWriter<>(
+        () ->
```

Comment:
I don't like how this part is formatted but I don't have a better way. Ideas welcome.